[Gdk pixbuf] Initial integration

ecalp-tps opened the pull request:
Hello,

I am a Software Engineering Intern at Google, working with the Third Party Security (TPS) team. As part of my internship I am working on fuzzing GNOME/freedesktop libraries and Gdk-Pixbuf is one of the libraries that we have picked for this work.

You can find the MR for integrating my fuzzers into the main Gdk-Pixbuf repository [here](https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/84). I've been in touch with the upstream and they are interested in integrating with OSS-Fuzz. However, there were some issues with my repo and their CI pipeline. I'm still waiting to hear from them to move forward but since my internship ends this week, I wanted to send in the PR. For now I’m fetching the Gdk-Pixbuf repo from my personal fork in the Dockerfile.

Thanks!
Ceyhun
ceyhunalp@google.com

jonathanmetzman commented on projects/gdk-pixbuf/build.sh:
```diff
@@ -0,0 +1,88 @@
+
+# Build glib
+pushd $WORK
+tar xvJf $SRC/glib-2.64.2.tar.xz
```
should probably do this in the Dockerfile.

jonathanmetzman reviewed (approved):
Can you link to your discussions with upstream?

jonathanmetzman commented:
Please fix msan fail or disable it.

jonathanmetzman commented:
Could you clarify if upstream has bandwidth to fix bugs?
If they dont then OSS-Fuzz will disclose bugs publically 90 days after they are found, even if they are unfixed.

inferno-chromium reviewed (changes requested):
Instead of personal fork, please use the upstream repo and add fuzzers in this repo here (in CL). Also, please disable memory sanitizers by listing sanitizers: field explicitly.

ecalp-tps commented:
> Can you link to your discussions with upstream?

https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/84#note_957852

ecalp-tps commented:
> Please fix msan fail or disable it.

Done.

ecalp-tps commented:
> Please fix msan fail or disable it.

Done.

ecalp-tps commented:
> Could you clarify if upstream has bandwidth to fix bugs?
> If they dont then OSS-Fuzz will disclose bugs publically 90 days after they are found, even if they are unfixed.

Here's the [response](https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/84#note_970953) from the upstream.

ecalp-tps commented:
> Instead of personal fork, please use the upstream repo and add fuzzers in this repo here (in CL). Also, please disable memory sanitizers by listing sanitizers: field explicitly.

Hello Abhishek, thank you for your help. I added the fuzzers here and updated the link in the Dockerfile to fetch the library from the upstream repo. I disabled the memory sanitizers and all build checks pass now.

inferno-chromium commented on projects/gdk-pixbuf/project.yaml:
```diff
@@ -0,0 +1,6 @@
+homepage: https://gitlab.gnome.org/GNOME/gdk-pixbuf/
+language: c
+primary_contact: jvoisin@google.com
```
Before we merge this, try getting maintainer email [ask them on those threads asap since your internship ending soon]. Julien's email should be in auto_ccs attribute. Same for the other 3 PRs. @Google-Autofuzz

ecalp-tps replied:
I moved Julien's email to auto_ccs and temporarily listed security-tps as the primary contact. I also asked upstream for an email address again so I'll update it as soon as they provide me with one.

ecalp-tps replied:
https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/84#note_971662

I sent them a message yesterday but still no word.

ecalp-tps replied:
I received an email address this evening. I updated my development [branch](https://github.com/ecalp-tps/oss-fuzz/commit/a92d26ad6b7560a0510363f5d99c35e926a93fd6#diff-13be7c9ab3b8b5460c1a8668e11ac3323ff1d091d83570e0b57fa020308b953c) with the address. Since I was still working on Gdk-Pixbuf and Cairo when their PRs were merged, it might be a good idea to revert them and send PRs from their development branches to get the most up-to-date version.

inferno-chromium replied:
This looks like a simple diff, why not just submit a new PR with adding the email address

inferno-chromium reviewed (approved):
Can you ask Julien to review this and other 2 PR as well.
